gravitee-am-management-api/gravitee-am-management-api-rest/src/main/java/io/gravitee/am/management/handlers/management/api/resources/organizations/environments/domains/TranslationsResource.java

```diff
@@ -0,0 +1,218 @@
+/**
+ * Copyright (C) 2015 The Gravitee team (http://gravitee.io)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.gravitee.am.management.handlers.management.api.resources.organizations.environments.domains;
+
+import io.gravitee.am.management.handlers.management.api.resources.AbstractResource;
+import io.gravitee.am.model.translation.Language;
+import io.gravitee.am.model.translation.Message;
+import io.gravitee.common.http.MediaType;
+import io.swagger.annotations.Api;
+import io.swagger.annotations.ApiOperation;
+import io.swagger.annotations.ApiResponse;
+import io.swagger.annotations.ApiResponses;
+
+import javax.validation.Valid;
+import javax.validation.constraints.NotNull;
+import javax.ws.rs.DELETE;
+import javax.ws.rs.GET;
+import javax.ws.rs.POST;
+import javax.ws.rs.PUT;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.container.AsyncResponse;
+import javax.ws.rs.container.Suspended;
+
+/**
+ * @author Eric LELEU (eric.leleu at graviteesource.com)
+ * @author GraviteeSource Team
+ */
+@Api(tags = {"translation"})
+public class TranslationsResource extends AbstractResource {
+
+    @GET
+    @Produces(MediaType.APPLICATION_JSON)
+    @ApiOperation(
+            nickname = "listLanguages",
+            value = "List languages supported for a security domain",
+            notes = "User must have the DOMAIN_TRANSLATIONS[LIST] permission on the specified domain " +
+                    "or DOMAIN_TRANSLATIONS[LIST] permission on the specified environment " +
+                    "or DOMAIN_TRANSLATIONS[LIST] permission on the specified organization.")
+    @ApiResponses({
+            @ApiResponse(code = 200, message = "List of languages for a security domain", response = Language.class, responseContainer = "List"),
+            @ApiResponse(code = 500, message = "Internal server error")})
+    public void list(
+            @PathParam("organizationId") String organizationId,
+            @PathParam("environmentId") String environmentId,
+            @PathParam("domain") String domain,
+            @Suspended final AsyncResponse response) {
+
+        //checkAnyPermission(organizationId, environmentId, domain, Permission.DOMAIN_TRANSLATIONS, Acl.LIST);
```

Comment:
Presumably these are commented out as placeholders?

Reply:
yes because the story is only about defining the REST API and the Data model.